src/openforms/prefill/base.py

```diff
@@ -14,9 +17,22 @@ def __contains__(self, thing):
         return True
 
 
-class BasePlugin(AbstractBasePlugin):
+class EmptyOptions(JsonSchemaSerializerMixin, serializers.Serializer):
+    pass
+
+
+class Options(TypedDict):
+    pass
+
+
+SerializerCls = type[serializers.Serializer]
+OptionsT = TypeVar("OptionsT", bound=Options)
+
+
+class BasePlugin(Generic[OptionsT], AbstractBasePlugin):
```

Comment:
I think with the new Python 3.12 syntax we can do this instead:

```suggestion
class BasePlugin[OptionsT: Options](AbstractBasePlugin):
```